doc/md/fundamentals/2-types/4-tuples.md

```diff
@@ -0,0 +1,108 @@
+---
+sidebar_position: 4
+---
+
+# Tuples
+
+A tuple is a fixed-size, ordered collection of values, where each element can have a different type. Tuples provide a way to group values by position, without the overhead of defining a record of named fields.  
+
+A tuple is grouped together in parentheses (`value1`, `value2`, `value3`). The type of a tuple is determined by types of its elements, such as ([`Text`](https://internetcomputer.org/docs/motoko/base/Text), [`Nat`](https://internetcomputer.org/docs/motoko/base/Nat), [`Bool`](https://internetcomputer.org/docs/motoko/base/Bool)). The values inside a tuple are evaluated in order from left to right. Tuples are immutable and their components cannot be modified after creation (unlike record fields, which can be declared mutable).  
+
+A tuple with zero elements is called the **unit value**, written as `()`. It represents a trivial result or token return value.  
+It's type is the **unit type**, also written `()`.  
+
+Unit values are typically used as dummy argument or return values for functions that don't require arguments, or don't produce results (but may produce side-effects).  
+
+## Defining a tuple
+
+```motoko
+let ghost = ("Motoko", 25);
+```
+
+The tuple's type is automatically inferred as `(Text, Nat)`, since `"Motoko"` is of type [`Text`](https://internetcomputer.org/docs/motoko/base/Text) and `25` is of type [`Nat`](https://internetcomputer.org/docs/motoko/base/Nat). 
+
+```motoko
+let ghost : (Text, Nat) = ("Motoko", 25);
+```
+
+:::info [Support for one length tuples]
+Motoko **does not** support length-one tuples. This is in contrast to languages such as Python, where a trailing comma differentiates a single-element tuple from a simple parenthesized value. In Motoko, `(x)` is always just `x`.
+:::
+
+## Accessing elements
+
+Elements are accessed using `.n` where `n` is the index (0-based indexing).
+
+```motoko no-repl
+// Ghost is a tuple of length 2
+let ghost : (firstName : Text, age : Nat) = ("Motoko", 25);
+let first = ghost.0; // "Motoko"
+let second = ghost.1; // 25
+```
+
+## Tuples as function return values
+
+Tuples are useful for returning multiple values from a function without requiring a separate data structure.
+
+```motoko no-repl
+func getUserInfo() : (Text, Nat) {
+    ("Ghost", 30);
+}
+
+getUserInfo();
+```
+
+## Nesting tuples
```

Comment:
I would move Nesting tuples to the end, since it relies on pattern matching of tuples.